skyportal/facility_apis/lco.py

```diff
@@ -0,0 +1,1032 @@
+                'target': target,
+                'acquisition_config': {},
+                'guiding_config': {'mode': 'OFF', 'optional': False},
+                'instrument_configs': [
+                    {
+                        'exposure_time': 60,
+                        'exposure_count': 1,
+                        'rotator_mode': 'VFLOAT',
+                        'optical_elements': {'slit': 'slit_1.6as'},
+                    }
+                ],
+            },
+            {
+                'type': 'LAMP_FLAT',
+                'instrument_type': '2M0-FLOYDS-SCICAM',
+                'constraints': constraints,
+                'target': target,
+                'acquisition_config': {},
+                'guiding_config': {'mode': 'OFF', 'optional': False},
+                'instrument_configs': [
+                    {
+                        'exposure_time': 50,
+                        'exposure_count': 1,
+                        'rotator_mode': 'VFLOAT',
+                        'optical_elements': {'slit': 'slit_1.6as'},
+                    }
+                ],
+            },
+        ]
+
+        tstart = request.payload["start_date"] + ' 00:00:00'
+        tend = request.payload["end_date"] + ' 00:00:00'
+
+        windows = [{'start': tstart, 'end': tend}]
+
+        altdata = request.allocation.altdata
+
+        # The full RequestGroup, with additional meta-data
+        requestgroup = {
+            'name': '%s' % (request.obj.id),  # The title
+            'proposal': altdata["PROPOSAL_ID"],
+            'ipp_value': request.payload["priority"],
+            'operator': 'SINGLE',
+            'observation_type': 'NORMAL',
+            'requests': [
+                {
+                    'configurations': configurations,
+                    'windows': windows,
+                    'location': location,
+                }
+            ],
+        }
+
+        return requestgroup
+
+
+class LCOAPI(FollowUpAPI):
+
+    """An interface to LCO operations."""
+
+    @staticmethod
+    def delete(request):
+
+        """Delete a follow-up request from LCO queue (all instruments).
+
+        Parameters
+        ----------
+        request: skyportal.models.FollowupRequest
+            The request to delete from the queue and the SkyPortal database.
+        """
+
+        from ..models import DBSession, FollowupRequest, FacilityTransaction
+
+        req = (
+            DBSession()
+            .query(FollowupRequest)
+            .filter(FollowupRequest.id == request.id)
+            .one()
+        )
+
+        altdata = request.allocation.altdata
+
+        if not altdata:
+            return
+
+        content = req.transactions[0].response["content"]
+        content = json.loads(content)
+        uid = content["id"]
+
+        r = requests.post(
+            'https://observe.lco.global/api/requestgroups/{}/cancel/'.format(uid),
+            headers={'Authorization': 'Token {}'.format(altdata["API_TOKEN"])},
```

Comment:
```suggestion
            f"https://observe.lco.global/api/requestgroups/{uid}/cancel/",
            headers={"Authorization": f'Token {altdata["API_TOKEN"]}'},
```
I think using the f formatting is preferable to format here for consistency with the rest of the codebase